### PR TITLE
Create MXNet Example

### PR DESCRIPTION
**Pull request by mariaboerner1987**
- create init
- create pyproject.toml
- create mxnet central
- create flower client
- create flower server

### CODE REVIEW

File: examples/mxnet_federated/server.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+"""Flower server example."""
+
+
```
Comment (tanertopal) [inline examples/mxnet_federated/server.py:3]:
```suggestion
```

File: examples/quickstart_mxnet/server.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+"""Flower server example."""
+
+
```
Comment (tanertopal) [inline examples/quickstart_mxnet/server.py:3]: Remove one new line
```suggestion
```

**Review by tanertopal — CHANGES_REQUESTED** [review]

File: examples/quickstart_mxnet/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+# MXNet: Basic Example
+
+This example demonstrates how to run a MXNEt machine learning project federated with Flower.
```
Comment (danieljanes) [inline examples/quickstart_mxnet/README.md:3]:
```suggestion
This example demonstrates how to run a MXNet machine learning project federated with Flower.
```

File: examples/mxnet_federated/README.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# MXNet: Basic Example
+
+This example demonstrates how an already existing centralized MXNEt-based machine learning project can be federated with Flower.
```
Comment (danieljanes) [inline examples/mxnet_federated/README.md:3]:
```suggestion
This example demonstrates how an already existing centralized MXNet-based machine learning project can be federated with Flower.
```

File: examples/mxnet_federated/README.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+This introductory example for Flower uses MXNet, but you're not required to be a MXNet expert to run the example. The example will help you to understand how Flower can be used to build federated learning use cases based on an existing MXNet projects.
```
Comment (danieljanes) [inline examples/mxnet_federated/README.md:5]:
```suggestion
This introductory example for Flower uses MXNet, but you're not required to be a MXNet expert to run the example. The example will help you to understand how Flower can be used to build federated learning use cases based on an existing MXNet project.
```

File: examples/mxnet_federated/README.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
+# MXNet: Basic Example
```
Comment (danieljanes) [inline examples/mxnet_federated/README.md:1]: Should we call this 
```suggestion
# MXNet: From Centralized To Federated
```
to be consistent with the PyTorch version?

Comment (danieljanes) [reply examples/mxnet_federated/README.md:1]: Which means we should also rename the directory to `mxnet_from_centralized_to_federated`

**Review by danieljanes — CHANGES_REQUESTED** [review]

File: examples/mxnet_from_centralized_to_federated/client.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        self, parameters: List[np.ndarray], config: Dict[str, str]
+    ) -> Tuple[int, float, float]:
+        # Set model parameters, evaluate model on local test dataset, return result
+        self.set_parameters(parameters)
+        loss, accuracy = mxnet_mnist.test(self.model, self.val_data, device=self.device)
+        return float(loss), self.val_data.batch_size, {"accuracy":float(accuracy)}
+
+
+def main() -> None:
+    """Load data, start CifarClient."""
```
Comment (danieljanes) [inline examples/mxnet_from_centralized_to_federated/client.py:62]: Typo:
```suggestion
    """Load data, start MNISTClient."""
```

File: examples/mxnet_from_centralized_to_federated/client.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+    # Setup context to GPU and if not available to CPU
```
Comment (danieljanes) [inline examples/mxnet_from_centralized_to_federated/client.py:64]:
```suggestion
    # Set context to GPU or - if not available - to CPU
```

File: examples/mxnet_from_centralized_to_federated/client.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    DEVICE = [mx.gpu() if mx.test_utils.list_gpus() else mx.cpu()]
+    # Load data
+    train_data, val_data = mxnet_mnist.load_data()
+    # Define model from centralized training
```
Comment (danieljanes) [inline examples/mxnet_from_centralized_to_federated/client.py:68]:
```suggestion
    # Load model (from centralized training)
```

File: examples/mxnet_from_centralized_to_federated/client.py [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+        mxnet_mnist.train(self.model, self.train_data, epoch=1, device=self.device)
+        return self.get_parameters(), self.train_data.batch_size, {}
+
+    def evaluate(
+        self, parameters: List[np.ndarray], config: Dict[str, str]
+    ) -> Tuple[int, float, float]:
+        # Set model parameters, evaluate model on local test dataset, return result
+        self.set_parameters(parameters)
+        loss, accuracy = mxnet_mnist.test(self.model, self.val_data, device=self.device)
+        return float(loss), self.val_data.batch_size, {"accuracy":float(accuracy)}
```
Comment (danieljanes) [inline examples/mxnet_from_centralized_to_federated/client.py:58]:
```suggestion
        return float(loss), self.val_data.batch_size, {"accuracy": float(accuracy)}
```

File: examples/mxnet_from_centralized_to_federated/client.py [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+        return param
+
+    def set_parameters(self, parameters: List[np.ndarray]) -> None:
+        # Collect model parameters and set new weight values
+        params = zip(self.model.collect_params(".*weight").keys(), parameters)
+        for key, value in params:
+            self.model.collect_params().setattr(key, value)
+
+    def fit(
+        self, parameters: List[np.ndarray], config: Dict[str, str]
```
Comment (danieljanes) [inline examples/mxnet_from_centralized_to_federated/client.py:45]:
```suggestion
        self, parameters: List[np.ndarray], config: Dict
```

File: examples/mxnet_from_centralized_to_federated/client.py [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+    ) -> Tuple[List[np.ndarray], int]:
+        # Set model parameters, train model, return updated model parameters
+        self.set_parameters(parameters)
+        mxnet_mnist.train(self.model, self.train_data, epoch=1, device=self.device)
+        return self.get_parameters(), self.train_data.batch_size, {}
+
+    def evaluate(
+        self, parameters: List[np.ndarray], config: Dict[str, str]
```
Comment (danieljanes) [inline examples/mxnet_from_centralized_to_federated/client.py:53]:
```suggestion
        self, parameters: List[np.ndarray], config: Dict
```

File: examples/mxnet_from_centralized_to_federated/client.py [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+    def set_parameters(self, parameters: List[np.ndarray]) -> None:
+        # Collect model parameters and set new weight values
+        params = zip(self.model.collect_params(".*weight").keys(), parameters)
+        for key, value in params:
+            self.model.collect_params().setattr(key, value)
+
+    def fit(
+        self, parameters: List[np.ndarray], config: Dict[str, str]
+    ) -> Tuple[List[np.ndarray], int]:
```
Comment (danieljanes) [inline examples/mxnet_from_centralized_to_federated/client.py:46]:
```suggestion
    ) -> Tuple[List[np.ndarray], int, Dict]:
```

File: examples/mxnet_from_centralized_to_federated/client.py [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+        # Set model parameters, train model, return updated model parameters
+        self.set_parameters(parameters)
+        mxnet_mnist.train(self.model, self.train_data, epoch=1, device=self.device)
+        return self.get_parameters(), self.train_data.batch_size, {}
+
+    def evaluate(
+        self, parameters: List[np.ndarray], config: Dict[str, str]
+    ) -> Tuple[int, float, float]:
```
Comment (danieljanes) [inline examples/mxnet_from_centralized_to_federated/client.py:54]:
```suggestion
    ) -> Tuple[int, float, Dict]:
```

File: examples/mxnet_from_centralized_to_federated/client.py [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+def main() -> None:
+    """Load data, start CifarClient."""
+
+    # Setup context to GPU and if not available to CPU
+    DEVICE = [mx.gpu() if mx.test_utils.list_gpus() else mx.cpu()]
+    # Load data
+    train_data, val_data = mxnet_mnist.load_data()
+    # Define model from centralized training
+    model = mxnet_mnist.model()
+    # Make one forward propagation to initialize parameters
```
Comment (danieljanes) [inline examples/mxnet_from_centralized_to_federated/client.py:70]:
```suggestion
    # Do one forward propagation to initialize parameters
```

File: examples/mxnet_from_centralized_to_federated/mxnet_mnist.py [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+        # Updates internal evaluation
+        metric.update(label, outputs)
+    print("validation acc: %s=%f" % metric.get())
+    print("validation loss:", loss)
+    accuracy = metric.get()[1]
+    return loss, accuracy
+
+
+def main():
+    # Setup context to GPU and if not available to CPU
```
Comment (danieljanes) [inline examples/mxnet_from_centralized_to_federated/mxnet_mnist.py:120]:
```suggestion
    # Set context to GPU or - if not available - to CPU
```

File: examples/quickstart_mxnet/README.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
+# MXNet: Basic Example
```
Comment (danieljanes) [inline examples/quickstart_mxnet/README.md:1]:
```suggestion
# Flower Example using MXNet
```
(to be consistent with our other quickstart examples)

**Review by danieljanes — CHANGES_REQUESTED** [review]

**Review by tanertopal — APPROVED** [review]

**Review by danieljanes — APPROVED** [review]

**Pull request merged**